### PR TITLE
tests: Setup E2E

### PR DESCRIPTION
**Pull request by krizzu**
## Summary

Adding Appium + WebDriverIO combo to run our E2E tests, instead of Detox. Closing #977 
Because it's too much effort now, E2E for iOS/Android won't be run on CI. Need to revisit this later.

### CODE REVIEW

File: example/__tests__/asyncstorage.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    // wait until content loads, as android's waitForSelectorTimeout setting does not seem to work
+    await new Promise((r) => setTimeout(r, 3000));
+    const el = await cmd.elementByLabel('functional-view');
+    await expect(await el.isExisting()).toEqual(true);
+  });
+
+  const testNames = Object.keys(testCases);
+  describe('storing / reading values', () => {
+    for (const name of testNames) {
+      it(`${name}`, async () => {
```
Comment (tido64) [inline example/__tests__/asyncstorage.spec.ts:15]:
```suggestion
      it(name, async () => {
```

File: example/__tests__/asyncstorage.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        const el = await cmd.elementByLabel(`test:${name}`);
+        await expect(await el.getText()).toEqual('Pass');
+      });
+    }
+  });
+
+  describe('storing / reading values with delegate', () => {
+    for (const currentName of testNames) {
+      const name = currentName + ' with delegate';
+      it(`${name}`, async () => {
```
Comment (tido64) [inline example/__tests__/asyncstorage.spec.ts:25]:
```suggestion
      it(name, async () => {
```

**Review by tido64 — APPROVED** [review]

**Comment by tido64** [comment]
Thanks for working on this ❤️

**Comment by krizzu** [comment]
Thanks! I did last minute changes for ts config, as something was not right with types. So I:
- Removed jest types and eslint config, as we use mocha now
- Swapped ts configs: tsconfig.json is now used as "all", to include example project too. It extends `tsconfig.build.json` which is used for release builds

File: example/__tests__/asyncstorage.spec.ts (hunk 1 above)
Comment (krizzu) [reply example/__tests__/asyncstorage.spec.ts]: This was done due to jest eslint rule. I removed it now.

**Pull request merged**